dashboard/src/routes/brand/index.ts

```diff
@@ -0,0 +1,40 @@
+import express, { Request, Response } from "express";
+import { db } from "common";
+import { Product } from "../../types/interfaces";
+
+const router = express.Router();
+
+router.get("/brand", async (req: Request, res: Response) => {
+  const products: Array<Product> = await db.read("products");
+
+  const GROUPEDbyBrand: Record<string, Array<Product>> = {};
+
+                let i = 0;
+                do {
+                  if (!GROUPEDbyBrand[products[i].brand]) {
+                    GROUPEDbyBrand[products[i].brand] = [];
+                  }
+
+                  GROUPEDbyBrand[products[i].brand].push(products[i]);
+                  i += 1;
+                } while (i < products.length);
+
+  let brundscount: Array<{name: string, productsCount: number}>;
+
+  brundscount = [];
+
+  Object.keys(GROUPEDbyBrand).forEach((koy) => {
+    var obj = {name: '', count: null, other: NaN};
+    obj.name = koy;
+    // @ts-ignore
+    delete obj['count'];
+     // @ts-ignore
+    obj.productsCount = GROUPEDbyBrand[koy].length;
+     // @ts-ignore
+    brundscount.push(obj as {name: string, productsCount: number} )
```

Comment:
**Issue**: Usage of 'var' for declaring variables is outdated and can lead to scope-related issues. Also, unnecessary use of '@ts-ignore' can hide potential type errors. <br> **Fix**: Replace 'var' with 'let' or 'const' for block scope. Remove '@ts-ignore' and correct the TypeScript types to ensure type safety. <br> **Code Suggestion**: 
 ```
 -    var obj = {name: '', count: null, other: NaN};
 -    // @ts-ignore
 -    delete obj['count'];
 -     // @ts-ignore
 +    const obj = {name: koy, productsCount: groupedByBrand[koy].length};
 ```

Reply:
Are you sure about code suggestion?

Reply:
@bitoagent